src/main/java/io/trino/plugin/tiledb/TileDBRecordCursor.java

```diff
@@ -447,11 +462,120 @@ private void setRanges(TileDBSplit split) throws TileDBError
+                    }
+                    else {
+                        if (isVar) {
+                            highBound = attBounds.getSecond().toString().getBytes();
+                            lowBound = attBounds.getFirst().toString().getBytes();
+                        }
+                        else {
+                            highBound = attBounds.getSecond();
+                            lowBound = attBounds.getFirst();
+                        }
+                        QueryCondition cond1 = new QueryCondition(ctx, att.getName(), lowBound, att.getType().javaClass(), TILEDB_GE);
+                        QueryCondition cond2 = new QueryCondition(ctx, att.getName(), highBound, att.getType().javaClass(), TILEDB_LE);
+                        query.setCondition(cond1.combine(cond2, TILEDB_AND));
+                    }
+                }
+                LOG.info("Query %s setting range for attribute %s to [%s, %s]", queryId, att.getName(), attBounds.getFirst(), attBounds.getSecond());
+            }
+        }
+        catch (Exception ex) {
+            ex.printStackTrace();
         }
-
         recordFunctionTime("setRanges", timer);
     }
 
+    /**
+     * Compute a (lower, upper) bound for the given attribute based on the given split.
+     */
+    private Pair getBoundsForAttribute(TileDBSplit split, Attribute attribute) throws TileDBError
+    {
+        if (emptyQueryAttributes.contains(attribute.toString())) {
+            return new Pair<>("", "");
+        }
+        String name = attribute.getName();
+        Pair<Long, Long> timer = startTimer();
+        TupleDomain<ColumnHandle> tupleDomain = split.getTupleDomain();
+        checkState(tupleDomain.getDomains().isPresent(), "No domains in tuple domain");
+
+        Domain domainPredicate = tupleDomain.getDomains().get().entrySet()
+                .stream()
+                .filter(e -> ((TileDBColumnHandle) e.getKey()).getColumnName().equals(name))
+                .map(Map.Entry::getValue)
+                .findFirst()
+                .orElse(null);
+
+        if (domainPredicate == null || domainPredicate.isAll()) {
+            return new Pair<>(null, null);
+        }
+
+        Object lowerBound = null;
+        Object upperBound = null;
+
+        try {
+            List<Range> orderedRanges = domainPredicate.getValues().getRanges().getOrderedRanges();
+            Class attType = attribute.getType().javaClass();
+
+            for (Range range : orderedRanges) {
+                Pair bounds = getBoundsForTrinoRange(range, attribute.getType());
+                lowerBound = bounds.getFirst() == null ? null : ConvertUtils.convert(bounds.getFirst(), attType);
+                upperBound = bounds.getSecond() == null ? null : ConvertUtils.convert(bounds.getSecond(), attType);
+            }
+        }
+        catch (NullPointerException exception) {
+            LOG.info("Attribute %s has no bounds", attribute.getName());
```

Comment:
Is it normal to have no bounds specified from trino?

Reply:
Yes, if no bounds are specified by the query, the range is [null, null]. Line 564 will throw a null pointer exception as there are no ranges. Instead of the try catch I could have an if condition though. Let me know what you think.

Reply:
👍